connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala

```diff
@@ -0,0 +1,140 @@
+package io.unitycatalog.connectors.spark
+
+import io.unitycatalog.client.{ApiClient, ApiException}
+import io.unitycatalog.client.api.TablesApi
+import io.unitycatalog.client.model.TableType
+
+import java.net.URI
+import java.util
+
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException
+import org.apache.spark.sql.catalyst.catalog.{CatalogStorageFormat, CatalogTable, CatalogTableType, CatalogUtils}
+import org.apache.spark.sql.connector.catalog._
+import org.apache.spark.sql.connector.expressions.Transform
+import org.apache.spark.sql.types.{DataType, StructField, StructType}
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+import scala.collection.JavaConverters._
+
+/**
+ * A Spark catalog plugin to get/manage tables in Unity Catalog.
+ */
+class UCSingleCatalog extends TableCatalog {
```

Comment:
Why is there single in the name?

Reply:
I'm open to better names. Unity Catalog is not a catalog, it's a metastore service that can manage many catalogs. I want to use the name `Single` to indicate that this plugin only maps to a single catalog in UC.